services/QuillLMS/app/models/unit_activity.rb

```diff
@@ -231,5 +234,8 @@ def self.get_classroom_user_profile(classroom_id, user_id)
           ua.id ASC
      SQL
     )
+
+    # note that we could also do this with a HAVING clause, but it is pretty difficult to read because the logic for computing those values is complex
+    data.filter { |ua| ua['finished'] || !ua['archived'] }
```

Comment:
I concur with your thought of doing this filter in Ruby rather than in SQL rather than making an already complicated query more complicated.